### DISCUSSION TITLE
Why does `Pod` not work with 2 generic-fields?

### DISCUSSION THREAD

**Discussion by DasLixou** [Lokathor/bytemuck#277] [Q&A] [answered]
I have a struct
```rs
struct MyStruct<T> {
    a: T,
    b: T,
}
```
and deriving Pod on it errors because multiple generics don't seem to be allowed, but I can't find any documentation of the reason.
So, stupidly asked, why?

**Comment by Lokathor** [comment, marked as answer]
at first glance it looks like that code should work? did you also use `repr(C)` in the actual code? if `repr(C)` is in the picture, then your type is basically the same as `[T; 2]`.

If that doesn't work then probably it's just that the proc-macro derive isn't sufficiently advanced, and someone could probably do a PR to support that. I'm going to make this into an issue so that this can be checked on longer term.

**Reply by DasLixou** [reply]
Yeah I did use repr(C) - I'll try to make a PR tomorrow, thanks for the info 😁 (but when I'm at it, why isn't it supported for multiple generics?)

**Reply by DasLixou** [reply]
So yeah, the error (with repr(C)) is
```
Pod requires cannot be derived for non-packed types containing generic parameters because the padding requirements can't be verified for generic non-packed structs
```

**Reply by DasLixou** [reply]
So @Lokathor I did some digging, the important lines seem to be 
https://github.com/Lokathor/bytemuck/blob/main/derive/src/traits.rs#L69-L78
The problem I have is that 
* A - `repr(C)` triggers "isn't packed" and `repr(packed)` triggers "must be either C or transparent"
* B - `repr(packed)` seems to be kinda bad, [the docs](https://doc.rust-lang.org/nomicon/other-reprs.html#reprpacked) say it is not to be used lightly as it could cause UB.
So my question is, what is the correct/allowed repr a struct like mine could have so I can change the macro accordingly?
Thanks

**Reply by Lokathor** [reply]
It should be possible when `T: Pod`, because Pod doesn't have padding. Perhaps I'm forgetting something there. I think the derive is just being too conservative.

**Reply by Lokathor** [reply]
http://github.com/Lokathor/bytemuck/issues/278 let's move further discussion to the issue instead

**Reply by DasLixou** [reply]
Oh yeah, didn't know you made an issue, sorry
